src/community/web-resources/src/main/java/org/geoserver/web/resources/PageResourceBrowser.java

```diff
@@ -0,0 +1,516 @@
+/* (c) 2016 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.web.resources;
+
+import org.apache.wicket.Component;
+import org.apache.wicket.ajax.AjaxRequestTarget;
+import org.apache.wicket.ajax.markup.html.AjaxLink;
+import org.apache.wicket.behavior.AttributeAppender;
+import org.apache.wicket.markup.html.basic.Label;
+import org.apache.wicket.markup.html.link.AbstractLink;
+import org.apache.wicket.markup.html.link.Link;
+import org.apache.wicket.model.Model;
+import org.apache.wicket.request.handler.resource.ResourceStreamRequestHandler;
+import org.apache.wicket.request.resource.ContentDisposition;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.geoserver.platform.resource.Paths;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.ResourceStore;
+import org.geoserver.platform.resource.Resources;
+import org.geoserver.web.GeoServerSecuredPage;
+import org.geoserver.web.treeview.TreeNode;
+import org.geoserver.web.treeview.TreeView;
+import org.geoserver.web.wicket.GeoServerDialog;
+import org.geoserver.web.wicket.ParamResourceModel;
+import org.geoserver.web.wicket.GeoServerDialog.DialogDelegate;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * The ResourceBrowser page.
+ * 
+ * @author Niels Charlier
+ *
+ */
+public class PageResourceBrowser extends GeoServerSecuredPage {
+    
+    private static final long serialVersionUID = 3979040405548783679L;
```

Comment:
Jody has asked me previously always to include a serialVersionUID and no @SuppressWarnings. I have even changed it in a lot of places during the code sprint, and then I found out Andrea agrees with you.

Wicket does serialise pages.